### PR TITLE
Policy separation

### PR DESCRIPTION
**Pull request by leonelmaia**
Separated policy  from SR using a interface to connect SR to the policy.

### CODE REVIEW

File: src/sbr/SR.java [hunk 1]
```diff
@@ -24,9 +24,10 @@
 
 	private HashMap<Vertex, Segment> segmentForVertex;
 	private HashMap<Vertex, Integer> subnetForVertex;
-
-	public SR(Graph graph) 
-	{
+	BidimensionalPolicy politica = new BidimensionalPolicy();
```
Comment (rmilfont) [inline src/sbr/SR.java]: Change this variable name. English must be used all the time, so never use Portuguese.

File: src/rsbr.java [hunk 2]
```diff
@@ -3,6 +3,7 @@
 
 import util.*;
 import rbr.*;
+import sbr.BidimensionalPolicy;
```
Comment (rmilfont) [inline src/rsbr.java]: Squash the "import" using *.

File: src/sbr/BidimensionalPolicy.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
+package sbr;
+
+import java.util.ArrayList;
+import util.Edge;
+import util.Graph;
+
+public class BidimensionalPolicy implements SBRPolicy {
```
Comment (rmilfont) [inline src/sbr/BidimensionalPolicy.java]: Change this name to "BidimensionalSBRPolicy".

File: src/sbr/BidimensionalPolicy.java [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+	
+	public BidimensionalPolicy() {
+		RRIndex = new int[2];
+		RRIndex[0] = -1;
+		RRIndex[1] = -1;
+		
+	}
+
+
+	/*
```
Comment (rmilfont) [inline src/sbr/BidimensionalPolicy.java]: Remove the comments.

File: src/sbr/BidimensionalPolicy.java [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+	private final static String[] RoundRobin = { "N", "E", "S", "W" };
+	private static int RRIndex[];
+	private ArrayList<Edge> edge;
+
+	BidimensionalPolicy(Graph graph){
+		edge = graph.getEdges();
+		RRIndex = new int[2];
+		RRIndex[0] = -1;
+		RRIndex[1] = -1;
+
```
Comment (rmilfont) [inline src/sbr/BidimensionalPolicy.java]: Remove this blank line.

File: src/sbr/BidimensionalPolicy.java [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+	BidimensionalPolicy(Graph graph){
+		edge = graph.getEdges();
+		RRIndex = new int[2];
+		RRIndex[0] = -1;
+		RRIndex[1] = -1;
+		
+	}
+	
+
```
Comment (rmilfont) [inline src/sbr/BidimensionalPolicy.java]: Remove this blank line.

File: src/sbr/BidimensionalPolicy.java [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+		RRIndex[1] = -1;
+		
+	}
+	
+	
+	public BidimensionalPolicy() {
+		RRIndex = new int[2];
+		RRIndex[0] = -1;
+		RRIndex[1] = -1;
+
```
Comment (rmilfont) [inline src/sbr/BidimensionalPolicy.java]: Remove this blank line.

File: src/sbr/BidimensionalPolicy.java [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+	}
+	
+	
+	public BidimensionalPolicy() {
+		RRIndex = new int[2];
+		RRIndex[0] = -1;
+		RRIndex[1] = -1;
+		
+	}
+
```
Comment (rmilfont) [inline src/sbr/BidimensionalPolicy.java]: Remove this blank line.

File: src/sbr/BidimensionalPolicy.java [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+			}
+		}
+		while (true) {
+			for (Edge ln : links) {
+				if (ln.color() == RoundRobin[index]) {
+					got = ln;
+					break;
+				}
+			}
+			if (got != null)
```
Comment (rmilfont) [inline src/sbr/BidimensionalPolicy.java]: Put the curly brackets for "if" statement.

File: src/sbr/BidimensionalPolicy.java [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+				break;
+			else {
+				if (RRIndex[1] == ((RRIndex[0] + 1) % 4))
```
Comment (rmilfont) [inline src/sbr/BidimensionalPolicy.java]: Put the curly brackets for "if" statement

File: src/sbr/BidimensionalPolicy.java [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+					index = (index + 3) % 4;
+				else
```
Comment (rmilfont) [inline src/sbr/BidimensionalPolicy.java]: Put the curly brackets for "else" statement.

File: src/sbr/BidimensionalPolicy.java [hunk 12]
```diff
@@ -0,0 +1,82 @@
...
+		}
+		return got;
+	}
+
+	public   void resetRRIndex() {
+		RRIndex[0] = -1;
+		RRIndex[1] = -1;
+	
+	}
+
```
Comment (rmilfont) [inline src/sbr/BidimensionalPolicy.java]: Remove these blank lines.

File: src/sbr/SBRPolicy.java [hunk 13]
```diff
@@ -0,0 +1,10 @@
+package sbr;
+
```
Comment (rmilfont) [inline src/sbr/SBRPolicy.java:2]: Remove this blank line.

File: src/sbr/SBRPolicy.java [hunk 14]
```diff
@@ -0,0 +1,10 @@
...
+import java.util.ArrayList;
+
```
Comment (rmilfont) [inline src/sbr/SBRPolicy.java]: Remove this blank line.

File: src/sbr/BidimensionalPolicy.java [hunk 15]
```diff
@@ -0,0 +1,82 @@
...
+			RRIndex[0] = RRIndex[1];
+			RRIndex[1] = index;
+		}
+		return got;
+	}
+
+	public   void resetRRIndex() {
+		RRIndex[0] = -1;
+		RRIndex[1] = -1;
+
```
Comment (rmilfont) [inline src/sbr/BidimensionalPolicy.java]: Remove this blank line.

File: src/sbr/BidimensionalPolicy.java [hunk 16]
```diff
@@ -0,0 +1,82 @@
+package sbr;
+
+import java.util.ArrayList;
+import util.Edge;
```
Comment (rmilfont) [inline src/sbr/BidimensionalPolicy.java]: Squash the "import" using *.

**Review by rmilfont — CHANGES_REQUESTED** [review]
Little changes in order to make the code clearer.

**Comment by rmilfont** [comment]
The conflict occurred because you didn´t follow the steps before push your code. Pay attention at https://github.com/LaNoC-UFC/GitHub-roadmap#7-after-pull-request-actions , case 1.

**Comment by rmilfont** [comment]
You didn´t set the commit template, please see: https://github.com/LaNoC-UFC/GitHub-roadmap#3-configure-the-commit-template , item 3.

**Comment by jmferreiratech** [comment]
Next time be sure to create a new branch instead of modifying the develop.

File: .gitignore [hunk 17]
```diff
@@ -1,9 +1,6 @@
-bin/
-out/
-.idea/
-.classpath
-.project
-*~
```
Comment (jmferreiratech) [inline .gitignore:6]: You shouldn't remove 'out/', '.idea' and '*~'.

File: src/sbr/SBRPolicy.java [hunk 18]
```diff
@@ -0,0 +1,10 @@
+package sbr;
+
+import java.util.*;
+
```
Comment (jmferreiratech) [inline src/sbr/SBRPolicy.java]: You can keep all imports together.

File: src/rsbr.java [hunk 19]
```diff
@@ -4,7 +4,8 @@
 import util.*;
 import rbr.*;
 import paths_gen.*;
-import sbr.SR;
+import sbr.*;
+
```
Comment (jmferreiratech) [inline src/rsbr.java:7]: Remove this extra line.

**Review by jmferreiratech — CHANGES_REQUESTED** [review]
Please try to always use curly braces, even with one-statement block.

**Review by jmferreiratech — APPROVED** [review]

**Pull request merged**